src/ast/bits.h

```diff
@@ -60,6 +62,43 @@ struct Bits {
     }
     WASM_UNREACHABLE();
   }
+
+  static Expression* makeSignExt(Expression* value, Index bytes, Module& wasm) {
```

Comment:
The threads proposal has new instructions for this: https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#new-sign-extending-operators

Reply:
Interesting. I guess when we add those to binaryen we can use them to replace `makeSignExt` etc.

Reply:
Binaryen already has them: https://github.com/WebAssembly/binaryen/pull/1167